### PR TITLE
Project outlining

### PR DESCRIPTION
**Pull request by PDempsey03**
This branch accomplishes a few things that should help with project outlining.
1. Server Handling
-You now have the ability for Unity to start up the Flask server when you run the game and end the server when the game is stopped

2. Define generator functionality
- Generators are what place objects in the world
- If you want to add something new to the world like clouds for example, you can make a CloudGenerator class which implements Generator.
- A generator class has two methods, Generate and Clear. Generate is self-explanatory and clear means deleting all generated objects from the soon to be previous world
- Scene should have a WorldGenerator class which contains all generators and handles their logic

3. AI output handling
- There is now a WorldInfo class which should be json serializable with the output of gemini.
- Doing this requires the class to be sent as part of the description

4. UI fix
- The new GeminiTerrain scene has a UI which properly scales to the screen size and aspect ratio
- the text box now also wraps so large amounts of text can be put into the text box while still seeing it all

5. Updated project structure
- Separated scenes and scripts into their own folders

### CODE REVIEW

File: ai_server/server.py [hunk 1]
```diff
@@ -12,15 +12,53 @@
 
 @app.route('/parse_description', methods=['POST'])
 def parse_description():
+
     description = request.json.get('description')
 
+    # class content should be a copy paste of the WorldInfo.cs file
```
Comment (WAlphonso2) [inline ai_server/server.py:18]: Wait this is stupid, just read the file into a string!!!

**Review by WAlphonso2 — CHANGES_REQUESTED** [review]
Close but not quite.

File: ai_server/server.py (hunk 1 above)
Comment (PDempsey03) [reply ai_server/server.py:18]: i thought of this too, but when you build the project say for a for webgl or just for a desktop application, the .cs file will be compiled and you wont be able to read the raw file anymore

Comment (WAlphonso2) [reply ai_server/server.py:18]: What if you just made the prompt smarter by giving it specific requirements instead of just handing it the entire code.

**Review by WAlphonso2 — APPROVED** [review]
LGTM

**Pull request merged**